src/hooks/useDebounce.js

```diff
@@ -0,0 +1,31 @@
+import { useCallback } from 'react';
+
+export const useDebounce = (func, waitAmount, immediate) => {
+  const debounceIt = () => {
+    // Closure timeout ref for all subsequent calls
+    let timeout;
+
+    // Return a func wrapper with the timeout var ref closure
+    return (...args) => {
+      const doItLater = () => {
+        timeout = null;
+        if (!immediate) func(...args);
```

Comment:
Why the `!immediate` condition? To prevent executing twice? If so, are there possible race conditions here?

Reply:
I dont see the race?

Reply:
I mean it's pretty unlikely, but the `immediate` conditional could be executed, then `immediate` set to `false`, then the function run (despite `immediate` being falsey at the time of execution).

Reply:
I'm not sure how the immediate is changed within the closure?